flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -123,7 +123,15 @@ public class ExecutionConfig implements Serializable, Archiveable<ArchivedExecut
             key("hidden.execution.mode")
                     .enumType(ExecutionMode.class)
                     .defaultValue(ExecutionMode.PIPELINED)
-                    .withDescription("Defines how data exchange happens - batch or pipelined");
+                    .withDescription(
+                            "Defines how data exchange happens "
+                                    + "- batch or pipelined(used in the deprecated DataSet API)");
+
+    private static final ConfigOption<String> JOB_TYPE =
```

Comment:
`jobType` is an automatically derived property instead of a user setting. Therefore, I would not put it in `ExecutionConfig`.
To achieve the goal to show jobType on the web UI, we can introduce a `jobType` to `JobConfigInfo` and `AccessExecutionGraph`.